web/src/components/SubmitAndRedirect.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useSubmitPersonalInfo } from '../hooks/useSubmitPersonalInfo';
+import PersonalInfoForm from './PersonalInfo/PersonalInfoForm';
+import Header from './Header/Header';
+
+const SubmitAndRedirect = () => {
+  const { submitPersonalInfo, loading, error, successMessage, cvData } = useSubmitPersonalInfo();
+  const navigate = useNavigate();
+
+  const handleFormSubmit = async (formData) => {
+    await submitPersonalInfo(formData);
+    if (cvData) {
+      navigate('/preview', { state: { cvData } });
+    }
+  };
+
+  return (
+    <>
+      <Header /> {/* ✅ Add header here */}
+      <PersonalInfoForm
+        onSubmit={handleFormSubmit}
+        loading={loading}
+        error={error}
+        successMessage={successMessage}
+      />
```

Comment:
Can you justify you decision to pass move the `useSubmitPersonalInfo` hook into the parent component? I think in this case it makes more sense to keep all the form logic located inside the form component. But happy to hear an argument against this